### ISSUE TITLE
Enforce privacy for leading underscore naming convention

### ISSUE THREAD

**Issue by lencioni** [brigade/scss-lint#721]
Although the `NameFormat` linter has an option to allow for leading underscores, it does not enforce the convention. It would be really great to either have a linter that does this, or teach `NameFormat` to do this.

What I mean by enforcing privacy for the leading underscore convention is, for variables, functions, and mixins that start with a leading underscore, we should verify that:

- any definitions are used within the same file
- any uses are defined within the same file

And potentially:

- any uses are preceded by their definitions

**Bad**
```scss
$_foo: red;

p {
  color: blue;
}
```

**Bad**
```scss
p {
  color: $_foo;
}
```

**Bad**
```scss
p {
  color: $_foo;
}

$_foo: red;
```

**Good**
```scss
$_foo: red;

p {
  color: $_foo;
}
```

And so on for functions and mixins.

**Comment by ljharb** [comment]
In addition it would be nice to make the prefix configurable.
